packages/components/src/autocomplete/index.js

```diff
@@ -409,78 +471,41 @@ export class Autocomplete extends Component {
 					}
 				);
 
-				if ( ! open ) {
+				if ( ! autocompleter ) {
 					this.reset();
 					return;
 				}
 
-				const safeTrigger = escapeRegExp( open.triggerPrefix );
+				const safeTrigger = escapeRegExp( autocompleter.triggerPrefix );
 				const match = text.match(
 					new RegExp( `${ safeTrigger }(\\S*)$` )
 				);
 				const query = match && match[ 1 ];
-				const {
-					open: wasOpen,
-					suppress: wasSuppress,
-					query: wasQuery,
-				} = this.state;
-
-				if (
-					open &&
-					( ! wasOpen ||
-						open.idx !== wasOpen.idx ||
-						query !== wasQuery )
-				) {
-					if ( open.isDebounced ) {
-						this.debouncedLoadOptions( open, query );
-					} else {
-						this.loadOptions( open, query );
-					}
-				}
-				// create a regular expression to filter the options
-				const search = open
-					? new RegExp( '(?:\\b|\\s|^)' + escapeRegExp( query ), 'i' )
-					: /./;
-				// filter the options we already have
-				const filteredOptions = open
-					? filterOptions(
-							search,
-							this.state[ 'options_' + open.idx ]
-					  )
-					: [];
-				// check if we should still suppress the popover
-				const suppress =
-					open && wasSuppress === open.idx ? wasSuppress : undefined;
-				// update the state
-				if ( wasOpen || open ) {
-					this.setState( {
-						selectedIndex: 0,
-						filteredOptions,
-						suppress,
-						search,
-						open,
-						query,
-					} );
-				}
-				// announce the count of filtered options but only if they have loaded
-				if ( open && this.state[ 'options_' + open.idx ] ) {
-					this.announce( filteredOptions );
-				}
+				this.setState( {
+					autocompleter,
+					AutocompleterUI:
```

Comment:
We store a component as state?

Reply:
yes, it depends on the "autocompleter" and if we generate it on each render it will remount on each render as well.

Reply:
Can we prevent re-renders somehow, at the moment when you type and nothing changes in the results, the popover walks in the direction of the caret position? In fact, it could always be rendered relative to the beginning of the phrase that triggers autocompletion.

Reply:
i think it's fixed.